### DISCUSSION TITLE
Do we need shared pages?

### DISCUSSION THREAD

**Discussion by falkoschindler** [zauberzeug/nicegui#158] [General]
The original idea of NiceGUI was to build an easy-to-use GUI framework that runs in a web browser. There was only one UI state shared across all connected clients. With added support for pages the need for "individual" or "private" pages came up. But the 3-line hello-world example should still be possible, which is why NiceGUI automatically creates an index page if there are UI elements in global scope. So beginners don't have to care about page decorators.

With upcoming NiceGUI version 1.0 we want to start building on FastAPI to make building [fully featured web applications](https://github.com/zauberzeug/nicegui/discussions/135) in Python even easier. Automatic index pages will remain. And individual pages will be the default - as they are already in the current version 0.9.25.

The question is: Do we still need _shared_ pages? Do we ever want to share the _UI_ state across browsers, e.g. seeing a dialog or drop-down menu magically opening for all connected clients? Or does the UI _bind_ to global state anyway, so that the UI elements themselves can be re-created per client?

What are your use-cases? @me21, you seem to work with shared pages (https://github.com/zauberzeug/nicegui/issues/155). Is it a core feature you're relying on or is switching to individual pages an option?

**Comment by me21** [comment]
Well, I have an UI for some app settings. I started it when the pages were shared by default, so I kept it that way so far.
I guess I can create UI elements bound to the same global state, and whenever the global state is changed by one client, it will still be reflected in other clients, even though the page itself is not shared?

**Comment by rodja** [comment]
Ok. In NiceGUI 1.0 we removed shared pages almost entirely. Only elements which are not written inside a page-decorated function are shared between clients.
